### ISSUE TITLE
Error installing thpp 

### ISSUE THREAD

**Issue by vislab2013** [facebook/fbcunn#66]
I'm getting this error when compiling thpp (tried on 2 separate machines with newly installed ubuntu 14.04, same result). Any sugestions how to work around this ?

[ 92%] Building CXX object test/CMakeFiles/tensor_serialization_test.dir/TensorSerializationTest.cpp.o
Linking CXX executable tensor_serialization_test
/usr/local/lib/libthrift.so: undefined reference to `boost::this_thread::interruption_point()'
/usr/local/lib/libthriftcpp2.so: undefined reference to `boost::this_thread::disable_interruption::disable_interruption()'
/usr/local/lib/libthriftcpp2.so: undefined reference to `boost::this_thread::disable_interruption::~disable_interruption()'
/usr/local/lib/libthrift.so: undefined reference to `boost::detail::get_current_thread_data()'
collect2: error: ld returned 1 exit status
make[2]: *** [test/tensor_serialization_test] Error 1
make[1]: *** [test/CMakeFiles/tensor_serialization_test.dir/all] Error 2
make: *** [all] Error 2

**Comment by mahyarnajibi** [comment]
I have the same problem. Ubuntu 14.04, libboost1.55, gcc 4.8.2. Any suggestions?

**Comment by mszlazak** [comment]
I had the same! Apparently there will be a fix.

**Comment by vislab2013** [comment]
Found a work around for this issue in: https://github.com/wb14123/thpp/commit/b194e1d4ca4db056593e44ba539fef2ff9e7f737

Just updated the thpp/CMakeLists.txt file in line 71 with the string SET(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-as-needed,-lboost_thread,-lboost_system") and everything else compiled without any issues, fbcunn included.

Cheers

**Comment by SalemAmeen** [comment]
Where can we find this file? please

**Comment by vislab2013** [comment]
you can find it under the folder thpp/thpp/CMakeLists.txt


Ofc this means that you'll have to install folly, fbthrift, thpp++ and fblualib by hand instead of using the script to automatically install those dependencies
curl -sk https://raw.githubusercontent.com/soumith/fblualib/master/install_all.sh | bash -e

Adding those flags worked for me, it should solve your problem aswell imo
Date: Mon, 22 Jun 2015 07:55:12 -0700

**Comment by arjenroodselaar** [comment]
This was related to https://github.com/facebook/fbthrift/issues/100. This has been fixed internally and a new release of fbthrift will go out soon.

**Comment by mszlazak** [comment]
Thank you!

**Comment by SalemAmeen** [comment]
Thank you!

**Comment by arjenroodselaar** [comment]
I pushed v0.28 of fbthrift yesterday which fixes this.

**Comment by mszlazak** [comment]
Great. Anything special we have to do or just follow the install directions as they are?

**Comment by arjenroodselaar** [comment]
It looks like the script in the instructions uses fbthrift master, which at this point should work.

**Comment by soumith** [comment]
sorry about this. I just fixed the fblualib install script so that this error does not happen now.
I made sure tagged versions of folly and fbthrift are installed, rather than trunk. I tested the install process on a fresh ubuntu 14.04 machine.

You'll have to redo the install process starting from this step:
https://github.com/facebook/fbcunn/blob/master/INSTALL.md#install-folly-fbthrift-thpp-and-fblualib
